File: codeiq_Q591/kato/root.rb
result = result_full = n = nil
loop do
  n = (n || 1) + 1
  i, f = Math.sqrt(n).to_s.split('.')
  result ||= n if f.chars[0..9].uniq.size == 10
  result_full ||= n if (i+f).chars[0..9].uniq.size == 10
  break if result && result_full
end
puts "only decimal place : #{result} / sqrt=#{Math.sqrt(result).to_s}"
puts "full number chars  : #{result_full} / sqrt=#{Math.sqrt(result_full).to_s}"


#=> only decimal place : 143 / sqrt=11.958260743101398
#=> full number chars  : 1362 / sqrt=36.90528417449187

File: codeiq_Q591/kato/root.php
<?php
$result = $result_full = $n = null;
function is_result($num)
{
    return count(array_unique(array_slice(str_split($num, 1), 0, 10))) == 10;
}
while (true) {
  $n = isset($n) ? $n + 1 : 1;
  $sqrt = explode('.', strval(sqrt($n)));
  if (count($sqrt) > 1) {
      list($i, $f) = $sqrt;
  } else {
      $i = $sqrt;       // 整数部のみなのでcontinueでも良さそう
      $f = '';
  }

  if (is_null($result) && is_result($f)) {
      $result = $n;
  }
  if (is_null($result_full) && is_result($i.$f)) {
      $result_full = $n;
  }
  if (isset($result) && isset($result_full)) break;
}
echo "only decimal place : {$result} / sqrt=".strval(sqrt($result))."\n";
echo "full number chars  : {$result_full} / sqrt=".strval(sqrt($result_full));

